### PR TITLE
Fix the EnemyWatcher trait (hotfix)

### PR DESCRIPTION
**Pull request by penev92**
Fixing two things that popped up in today's test game with @obrakmann and @pchote - notification intervals and players that hear the notifications. Currently in a muliplayer game you hear everyone's notifications, not just yours.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs [hunk 1]
```diff
@@ -102,7 +102,7 @@ void Announce(Actor self, TraitPair<AnnounceOnSeen> announce)
 		{
 			// Audio notification
 			if (self.World.LocalPlayer != null)
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs]: This should either become `self.Owner != null` then or go away.

**Comment by Mailaender** [comment]
:+1:

**Comment by Mailaender** [comment]
Actually not. I still see worm sign discovery beacons from enemy players.

File: OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs [hunk 2]
```diff
@@ -101,8 +101,7 @@ public void Tick(Actor self)
 		void Announce(Actor self, TraitPair<AnnounceOnSeen> announce)
 		{
 			// Audio notification
-			if (self.World.LocalPlayer != null)
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs]: I think you'd need to undo this change and then change this check to `if (self.World.LocalPlayer == self.Owner)`

Comment (penev92) [inline OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs]: To what end?

Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs]: To fix the issue Mailaender noted in his last comment. You'd need to wrap the whole body of this method beneath that check.

**Comment by penev92** [comment]
Fixed so both the "wormsign" and worm attack radar pings show only to the correct players.

File: OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs [hunk 3]
```diff
@@ -23,7 +23,7 @@ class EnemyWatcherInfo : ITraitInfo
 		public readonly int ScanInterval = 25;
 
 		[Desc("Minimal interval in ticks between notifications.")]
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs]: Since you're here could you make this `Minimal ticks in-between notifications.`?

File: OpenRA.Mods.D2k/Activities/SwallowActor.cs [hunk 4]
```diff
@@ -93,7 +93,9 @@ void PlayAttackAnimation(Actor self)
 		void NotifyPlayer(Player player, WPos location)
 		{
 			Sound.PlayNotification(player.World.Map.Rules, player, "Speech", swallow.Info.WormAttackNotification, player.Country.Race);
-			radarPings.Add(() => true, location, Color.Red, 50);
+			
+			if (player == player.World.LocalPlayer)
```
Comment (Phrohdoh) [inline OpenRA.Mods.D2k/Activities/SwallowActor.cs]: Would the spectator want to see this? If so, `World.RenderPlayer`.

**Comment by Phrohdoh** [comment]
Travis failed with `OpenRA.Mods.D2k/Activities/SwallowActor.cs:L96: [SP2000] Invalid spacing at the end of the line.`.

**Comment by obrakmann** [comment]
This looks good now and should do the trick. Singleplayer works fine as before. I'd like to have a quick test game on this branch before it gets merged, though. Otherwise, :+1:

**Comment by penev92** [comment]
At this point this needs either @Mailaender to reconfirm his +1, or someone else to review it, sorry.

**Comment by Mailaender** [comment]
I am still seeing red triangles without sound (maybe worm attacks?) in the unexplored shroud when playing with a lot of AI bots.

**Comment by abcdefg30** [comment]
Didn't notice anything bad while the testgame with @penev92 and @obrakmann. :+1: / :white_check_mark:

**Comment by penev92** [comment]
I was unable to reproduce @Mailaender's problem in a 4-team match with me and 7 bots on [this map](http://i.imgur.com/NN5x1RX.png)

File: OpenRA.Mods.D2k/Activities/SwallowActor.cs [hunk 5]
```diff
@@ -113,11 +115,7 @@ public override Activity Tick(Actor self)
 				if (self.World.SharedRandom.Next() % 100 <= sandworm.Info.ChanceToDisappear)
 				{
 					self.CancelActivity();
-					self.World.AddFrameEndTask(w => w.Remove(self));
-
-					var wormManager = self.World.WorldActor.TraitOrDefault<WormManager>();
-					if (wormManager != null)
-						wormManager.DecreaseWormCount();
+					self.World.AddFrameEndTask(w => self.Kill(self));
```
Comment (Mailaender) [inline OpenRA.Mods.D2k/Activities/SwallowActor.cs]: `self.Destroy()` is probably what you want here as the worm does not suicide.
